packages/frontend/src/hooks/dashboard/useDashboardRefresh.tsx

```diff
@@ -1,27 +1,39 @@
-import { useIsFetching, useQueryClient } from 'react-query';
+import { useCallback, useMemo } from 'react';
+import { Query, useIsFetching, useQueryClient } from 'react-query';
+
+const QUERIES_TO_REFRESH = [
+    'savedChartResults',
+    'saved_query',
+    'saved_dashboard_query',
+    'dashboards',
+];
+
+const queryPredicate = (query: Query) => {
+    return QUERIES_TO_REFRESH.some((key) => {
+        const firstQueryKey =
+            typeof query.queryKey === 'string'
+                ? query.queryKey
+                : query.queryKey[0];
```

Comment:
this is because there's no partial query key matching when adding a `predicate` right?